src/signal_handler.rs

```diff
@@ -0,0 +1,154 @@
+use super::*;
+
+use command_group::{CommandGroup, GroupChild};
+use signal_hook::consts::signal::*;
+use signal_hook::consts::TERM_SIGNALS;
+use signal_hook::flag;
+use signal_hook::iterator::exfiltrator::WithOrigin;
+use signal_hook::iterator::SignalsInfo;
+use std::io::Read;
+use std::process::Command;
+use std::sync::{atomic::AtomicBool, Arc};
+
+pub(crate) struct SignalHandler {
+  verbosity: Verbosity,
+  signals: Option<SignalsInfo<WithOrigin>>,
+}
+
+impl SignalHandler {
+  pub(crate) fn install(verbosity: Verbosity) -> Result<(), std::io::Error> {
+    let mut instance = Self::instance();
+    instance.verbosity = verbosity;
+
+    let signal_flag = Arc::new(AtomicBool::new(false));
+
+    const MISC_SIGNALS: &[i32] = &[
+      #[cfg(not(windows))]
+      SIGHUP,
+      // SIGABRT,
+      // SIGPIPE,
+      // SIGALRM,
+    ];
+
+    let mut sigs = MISC_SIGNALS.to_vec();
+    sigs.extend(TERM_SIGNALS);
+
+    let signals = SignalsInfo::<WithOrigin>::new(&sigs)?;
+    instance.signals = Some(signals);
```

Comment:
```suggestion
    instance.signal_info = Some(SignalsInfo::<WithOrigin>::new(&signals)?);
```

Reply:
Are we unconditionally setting this on the instance every time? Could we instead set it once in the `instance` method?